tests/integration/scan-family-consistency.tcl

```diff
@@ -29,30 +49,23 @@ test {scan family consistency with configured hash seed} {
                 } else {
                     fail "replica did not catch up dbsize (primary=[$primary dbsize], replica=[$replica dbsize])"
                 }
-                set cursor {{0} {}}
-                while {1} {
-                    set primary_cursor_next [$primary scan [lindex $cursor 0]]
-                    set replica_cursor_next [$replica scan [lindex $cursor 0]]
-                    assert_equal $primary_cursor_next $replica_cursor_next
-                    if {[lindex $primary_cursor_next 0] eq "0"} {
-                        assert_equal "0" [lindex $replica_cursor_next 0]
-                        break
-                    }
-                    set cursor $primary_cursor_next
-                }
+
+                set keys [scan_interleaved $primary $replica scan]
+                set keys [lsort -unique $keys]
+                assert_equal [expr {$n+3}] [llength $keys]
 
                 foreach {cmd key} {hscan h sscan s zscan z} {
-                    set cursor {{0} {}}
-                    while {1} {
-                        set primary_cursor_next [$primary $cmd $key [lindex $cursor 0]]
-                        set replica_cursor_next [$replica $cmd $key [lindex $cursor 0]]
-                        assert_equal $primary_cursor_next $replica_cursor_next
-                        if {[lindex $primary_cursor_next 0] eq "0"} {
-                            assert_equal "0" [lindex $replica_cursor_next 0]
-                            break
+                    set keys [scan_interleaved $primary $replica $cmd $key]
+
+                    if {$cmd eq "hscan" || $cmd eq "zscan"} {
+                        set extracted_keys {}
+                        foreach {k v} $keys {
+                            lappend extracted_keys $k
                         }
```

Comment:
It is possible to use HSCAN NOVALUES and ZSCAN NOSCORES to make these commands return only the "keys" (field names, elements) and avoid this extract logic.

Idea (untested):

```tcl
                foreach {cmd key extra} {hscan h {novalues} sscan s {} zscan z {noscores}} {
                    set items [scan_interleaved $primary $replica $cmd $key {*}$extra]
```